pandas/tests/generic/test_generic.py

```diff
@@ -612,6 +612,16 @@ def test_pct_change(self, periods, fill_method, limit, exp):
         else:
             tm.assert_series_equal(res, Series(exp))
 
+    @pytest.mark.parametrize("DF,idx,first_idx,last_idx", [
+        ({'A': [1, 2, 3]}, [1, 1, 2], 1, 2),
+        ({'A': [1, 2, 3]}, [1, 2, 2], 1, 2),
+        ({'A': [1, 2, 3, 4]}, ['d', 'd', 'd', 'd'], 'd', 'd')])
+    def test_valid_index(self, DF, idx, first_idx, last_idx):
+        # GH 21441
+        df1 = pd.DataFrame(DF, index=idx)
```

Comment:
You can just cal this `df`; there's no ambiguity since there's only one frame in the test.  Also `DataFrame` is imported, so the `pd.` isn't needed.